libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1188,7 +1188,8 @@ string ABIFunctions::abiDecodingFunctionCalldataArray(ArrayType const& _type)
 	solAssert(_type.dataStoredIn(DataLocation::CallData), "");
 	if (!_type.isDynamicallySized())
 		solAssert(_type.length() < u256("0xffffffffffffffff"), "");
-	solAssert(!_type.baseType()->isDynamicallyEncoded(), "");
+	if (_type.baseType()->isDynamicallyEncoded())
+		solUnimplemented("Calldata arrays with non-value base types are not yet supported by Solidity.");
```

Comment:
Not sure whether ``by Solidity`` is actually needed, but in general the change looks fine.

Reply:
I wanted to make the distinction between "not supported by ABIEncoderV2" or "by Solidity as a whole".